lib/Smartcat/App/Command/push.pm

```diff
@@ -208,7 +226,37 @@ sub update {
         "No documents for files:" . join( ', ', @files_without_documents ) )
       if @files_without_documents;
 
-    $api->update_document( @{ $lang_pairs{$_} } ) for ( keys %lang_pairs );
+    for ( keys %lang_pairs ) {
+        my @arr = @{ $lang_pairs{$_} };
+        
+        my $path = $arr[0];
+        my $document_id = $arr[1];
+
+        $api->update_document( $path, $document_id );
+        
+        if ( $rundata->{split_filename_by_triple_dash} ) {
+            my $file_name = get_file_name(
+                $path,
+                $rundata->{filetype},
+                $target_languages[0]);
+            my $document_name = $arr[2];
+
+            # print "\tfile_name: " .$file_name."\n";
+            # print "\tdocument_name: " .$document_name."\n";
+
+            if ($file_name !~ $document_name) {
```

Comment:
I suggest not using rexep matching (`!~`). Instead, use string comparison (`ne`, meaning _not equal_).

Reply:
Changed: `if ($file_name ne $document_name) {...}`